realhf/impl/model/nn/real_llm_api.py

```diff
@@ -146,8 +185,18 @@ def post_process(self):
         return None
 
     @property
-    def share_embeddings_and_output_weights(self):
-        return self.config.share_embeddings_and_output_weights
+    def shared_embedding_or_output_weights(self) -> bool:
+        return self.config.tied_embedding
+
+    def shared_embedding_or_output_weight(self) -> None | torch.Tensor:
```

Comment:
Maybe rename one of them? They are too similar.

Reply:
Reserve `shared_embedding_or_output_weight` in consistent with Megatron-LM. `shared_embedding_or_output_weights` is removed since we do not use Megatron's `finalize_grad` func.